site/en/docs/extensions/mv3/getstarted/tut-reading-time/index.md

```diff
@@ -0,0 +1,304 @@
+`content.js`.
+
+{% Label %}manifest.json:{% endLabel %}
+
+```json
+{
+  ...
+  "content_scripts": [
+      {
+        "js": ["scripts/content.js"],
+        "matches": [
+          "https://developer.chrome.com/docs/extensions/*",
+          "https://developer.chrome.com/docs/webstore/*"
+        ]
+      }
+    ]
+}
+```
+
+The `“matches”` field can have one or more [match patterns][doc-match]. These allow the browser to
+identify which sites to inject the content scripts into. Match patterns consist of three parts
+`<scheme>://<host><path>`. They can contain '`*`' characters.
+
+{% Details %}
+{% DetailsSummary %}
+💡 **Does this extension display a permission warning?**
+{% endDetailsSummary %}
+
+When a user installs an extension, the browser informs them what the extension has the capacity to
+do. Content scripts request permission to run on the sites that meet the match pattern criteria. 
+
+In this example, the user would see the following permission warning:
+<figure>
+{% Img src="image/BhuKGJaIeLNPW9ehns59NfwqKxF2/rKDdOyri9x8VkhTEXbO6.png", alt="Permission warning the user will see when installing the Reading time extension", width="338", height="175", class="screenshot" %}
+  <figcaption>
+  Reading time permission warning. 
+  </figcaption>
+</figure>
+
+To dive deeper on extension permissions, see [Declaring permissions and warn users][doc-perms]. 
+
+{% endDetails %}
+
+### Step 4: Calculate and insert the reading time {: #step-4 }
+
+Content scripts can use the standard [Document Object Model][mdn-dom] (DOM) to read and change the
+content of a page. The extension will first check if the page contains the `<article>` element.
+Then, it will count all the words within this element and create a paragraph that displays the total
+reading time.
+
+Create a new folder called `scripts`. Add a file called `content.js` and include the following code: 
+
+{% Label %}content.js:{% endLabel %}
+
+```js
+const article = document.querySelector("article");
+
+// `document.querySelector` may return null if the selector doesn't match anything.
+if (article) {
+  const text = article.textContent;
+  const wordMatchRegExp = /[^\s]+/g; // Regular expression
+  const words = text.matchAll(wordMatchRegExp);
+  // matchAll returns an iterator, convert to array to get word count
+  const wordCount = [...words].length;
+  const readingTime = Math.round(wordCount / 200);
+  const badge = document.createElement("p");
+  // Use the same styling as the publish information in an article's header
+  badge.classList.add("color-secondary-text", "type--caption");
+  badge.textContent = `⏱️ ${readingTime} min read`;
+
+  // Support for API reference docs
+  const heading = article.querySelector("h1");
+  // Support for article docs with date
+  const date = article.querySelector("time")?.parentNode;
+
+  (date ?? heading).insertAdjacentElement("afterend", badge);
+}
+```
+
+{% Details %}
+{% DetailsSummary %}
+💡 **Interesting JavaScript used in this code**
```

Comment:
Might also want to mention the use of [optional chaining](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining) (`?.`) and the [nullish coalescing operator](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Nullish_coalescing_operator) (`??`).

Reply:
Brilliant! Added to commit.